src/sys/socket/mod.rs

```diff
@@ -241,6 +247,19 @@ impl SockProtocol {
     #[cfg(apple_targets)]
     #[allow(non_upper_case_globals)]
     pub const KextEvent: SockProtocol = SockProtocol::Icmp; // Matches libc::SYSPROTO_EVENT
+
+    /// Packet filter on IPv4 traffic
+    // NOTE: placed here due to conflict (little endian arch) with SockProtocol::NetLinkISCI
+    #[cfg(linux_android)]
+    #[allow(non_upper_case_globals)]
+    #[cfg(target_endian = "little")]
+    pub const EthIp: SockProtocol = unsafe { std::mem::transmute::<i32, SockProtocol>((libc::ETH_P_IP as u16).to_be() as i32) };
+
+    /// Packet filter on IPv4 traffic
+    #[cfg(linux_android)]
+    #[allow(non_upper_case_globals)]
+    #[cfg(target_endian = "big")]
+    pub const EthIp: i32 = (libc::ETH_P_IP as u16).to_be() as i32;
```

Comment:
Sorry for being unclear, this should be an enum variant

Reply:
On big-endian targets, no enum variant has value 2048, so that `std::mem::transmute()` is UB, which is the reason why I said we should make it an enum variant

Reply:
yes it should work now